cli/module_build.go

```diff
@@ -437,13 +437,81 @@ func reloadModuleAction(c *cli.Context, vc *viamClient) error {
 				return err
 			}
 		}
+		if !c.Bool(moduleFlagLocal) {
+			if manifest == nil || manifest.Build == nil || manifest.Build.Path == "" {
+				return errors.New(
+					"remote reloading requires a meta.json with the 'build.path' field set. " +
+						"try --local if you are testing on the same machine.",
+				)
+			}
+			if err := validateReloadableArchive(c, manifest.Build); err != nil {
+				return err
+			}
+			if err := addShellService(c, vc, part.Part, true); err != nil {
+				return err
+			}
+			infof(c.App.Writer, "copying %s to part %s", manifest.Build.Path, part.Part.Id)
+			err = vc.copyFilesToFqdn(
+				part.Part.Fqdn, c.Bool(debugFlag), false, false, []string{manifest.Build.Path},
+				reloadingDestination(c, manifest), logging.NewLogger("reload"))
+			if err != nil {
+				return err
+			}
+		}
 		needsRestart, err = configureModule(c, vc, manifest, part.Part)
 		if err != nil {
 			return err
 		}
 	}
 	if needsRestart {
-		return restartModule(c, vc, part.Part, manifest)
+		err = restartModule(c, vc, part.Part, manifest)
+		if s, ok := status.FromError(err); ok && s.Code() == codes.NotFound {
+			warningf(c.App.ErrWriter,
+				"viam-server couldn't find your module to restart it; "+
+					"this happens if the module couldn't start, and may not indicate an error.",
```

Comment:
> "this happens if the module couldn't start, and may not indicate an error."

So if `NotFound` is returned, then the RDK didn't have the right module running. Is this error saying that if that happened, the module probably failed to start in the first place? And, why would that indicate no error?

Reply:
oops will remove this -- this is caused by RSDK-7513, which I hadn't diagnosed yet when I wrote the error

Reply:
done in 4358a444a